src/gui/transferlistsortmodel.cpp

```diff
@@ -86,179 +86,157 @@ void TransferListSortModel::disableTrackerFilter()
+        }
+        break;
+
+    case TransferListModel::TR_AMOUNT_DOWNLOADED:
+    case TransferListModel::TR_AMOUNT_DOWNLOADED_SESSION:
+    case TransferListModel::TR_AMOUNT_LEFT:
+    case TransferListModel::TR_AMOUNT_UPLOADED:
+    case TransferListModel::TR_AMOUNT_UPLOADED_SESSION:
+    case TransferListModel::TR_COMPLETED:
+    case TransferListModel::TR_ETA:
+    case TransferListModel::TR_LAST_ACTIVITY:
+    case TransferListModel::TR_SIZE:
+    case TransferListModel::TR_TIME_ELAPSED:
+    case TransferListModel::TR_TOTAL_SIZE:
+        {
+            const auto left = leftValue.toLongLong();
+            const auto right = rightValue.toLongLong();
+
+            if ((left < 0) && (right < 0))
+                break;
+            else if (right < 0)
+                return true;
+            else if (left < 0)
+                return false;
+            else if (left != right)
+                return left < right;
+        }
+        break;
+
+    case TransferListModel::TR_AVAILABILITY:
+    case TransferListModel::TR_PROGRESS:
+    case TransferListModel::TR_RATIO:
+    case TransferListModel::TR_RATIO_LIMIT:
+        {
+            const auto left = leftValue.toReal();
+            const auto right = rightValue.toReal();
+
+            if ((left < 0) && (right < 0))
+                break;
+            else if (right < 0)
+                return true;
+            else if (left < 0)
+                return false;
+            else if (left != right)
+                return left < right;
+        }
+        break;
 
     case TransferListModel::TR_STATUS:
         {
-            const auto stateL = leftValue.value<BitTorrent::TorrentState>();
-            const auto stateR = rightValue.value<BitTorrent::TorrentState>();
+            const auto left = leftValue.value<BitTorrent::TorrentState>();
+            const auto right = rightValue.value<BitTorrent::TorrentState>();
 
-            if (stateL != stateR)
-                return stateL < stateR;
-            return invokeLessThanForColumn(TransferListModel::TR_QUEUE_POSITION);
+            if (left != right)
+                return left < right;
         }
+        break;
 
     case TransferListModel::TR_ADD_DATE:
     case TransferListModel::TR_SEED_DATE:
     case TransferListModel::TR_SEEN_COMPLETE_DATE:
         {
-            const auto dateL = leftValue.toDateTime();
-            const auto dateR = rightValue.toDateTime();
-
-            if (dateL.isValid() && dateR.isValid())
-            {
-                if (dateL != dateR)
-                    return dateL < dateR;
-            }
-            else if (dateL.isValid())
-            {
-                return true;
-            }
-            else if (dateR.isValid())
-            {
-                return false;
-            }
+            const auto left = leftValue.toDateTime();
+            const auto right = rightValue.toDateTime();
 
-            return hashLessThan();
+            if (left != right)
```

Comment:
How about "invalid" dates?

Reply:
It is delegated to the comparator < of QDateTime class.